### PR TITLE
Quick fix on handle_mapping. arrval should be used instead of retval.

### PR DESCRIPTION
**Pull request by jabiinfante**
This PR should fix this bug: https://bugs.php.net/bug.php?id=71484

### CODE REVIEW

File: tests/bug_parsing_alias.phpt [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+<?php
+
+$yaml_code = <<<YAML
+basic:
+  - &external {value: 22 }
+keylevel0:
+  keylevel1: 
+    value: 1
+  keylevel1.1:
+    keylevel2: #&strange
```
Comment (ranvis) [inline tests/bug_parsing_alias.phpt]: Doesn't this need uncommenting &strange?

**Comment by ranvis** [comment]
Other than that it looks good to me to fix enbug in #24...

**Comment by ranvis** [comment]
FWIW the test yaml_002.phpt is failing.
I assume this has nothing to do with this PR but because PHP now compares datetime fractional seconds on == operator.
php/php-src@58797219362e1ced (php-7.0.4, php-5.6.19)
So I still think this PR good to merge.

**Comment by bd808** [comment]
Thanks for the PR @jabiinfante. I apologize for letting it sit so long without any input.

**Pull request merged**